### PR TITLE
Update README to reflect wider scope of repository

### PR DESCRIPTION
**Pull request by hanno-becker**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,53 +1,57 @@
...
-- Optimal code is target-specific, but a large variety of CPU-specific implementations makes a library harder to both use and maintain.
-- Optimal code is complex (e.g. relying on handwritten assembly and extensive interleaving), impeding maintainenance and
-  amenability for audit, and potentially hardening verification efforts.
+### Goals
 
-In doubt, **mlkem-c-aarch64** chooses assurance and ease of use over performance: We only include implementations into **mlkem-c-aarch64** which are manually auditable or (ideally _and_) for which we see a path towards formal verification. All assembly should be as readable as possible and micro-optimization ideally deferred to automated tooling such as [SLOTHY](https://slothy-optimizer.github.io/slothy/). Ultimately, **mlkem-c-aarch64** strives for constant-time implementations for which the C-code is, at minimum, verified to be free of undefined behaviour, and where all assembly is functionally verified.
+**mlkem-native** aims for _assurance_, _ease of use_, and _performance_. In doubt, assurance and ease of
+use take precedence over performance: We only include implementations into **mlkem-c-aarch64** which are manually
```
Comment (mkannwischer) [inline README.md:17]:
```suggestion
use take precedence over performance: We only include implementations into **mlkem-native** which are manually
```

File: mk/config.mk [hunk 2]
```diff
@@ -29,6 +29,7 @@ CFLAGS += \
 	-Wextra \
 	-Wpedantic \
 	-Werror \
+        -std=c99 \
```
Comment (mkannwischer) [inline mk/config.mk:32]: this change should not be in this PR?

**Review by mkannwischer — APPROVED** [review]
Please remove the change to config.mk; otherwise this looks good to me.

File: mk/config.mk (hunk 2 above)
Comment (hanno-becker) [inline mk/config.mk:32]:
```suggestion
```

**Pull request merged**